### PR TITLE
Allow specification of index size in MB per day

### PR DESCRIPTION
**Pull request by gravesb**
To make the calculation of maxTotalDataSizeMB a little more intuitive, this allows a value in MB per day to be specified in the data bag and the maxTotal index size will be calculated based on that value along with the other cluster size values.

### CODE REVIEW

File: recipes/_configure_indexes.rb [hunk 1]
```diff
@@ -36,6 +36,20 @@
       :index
     end
 
+  daily_mb = hash.delete('_maxDailyDataSizeMB')
```
Comment (nmattam) [inline recipes/_configure_indexes.rb]: Looks like we are assuming that this value should include some padding? If so should the README update say that?

Comment (gravesb) [inline recipes/_configure_indexes.rb]: I'm not sure I understand your question.  What do you mean it should include padding?

File: recipes/_configure_indexes.rb [hunk 2]
```diff
@@ -36,6 +36,20 @@
...
+  if [:index, :default].include?(stanza_type) && daily_mb && !hash.key?('maxTotalDataSizeMB')
+    settings = CernerSplunk.my_cluster_data(node).fetch('settings', {})
+    replication_factor = settings['replication_factor'] || 1
+    indexer_count = settings['_cerner_splunk_indexer_count'] || 1
+
+    frozen_time_in_days = hash['frozenTimePeriodInSecs'] || config.fetch('default', {})['frozenTimePeriodInSecs']
```
Comment (nmattam) [inline recipes/_configure_indexes.rb]: Should the variable be _frozen_time_in_secs_ on line 45 and 47 ?

File: docs/databags.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ The Cluster Hash is part of a plaintext data bag item that defines a logical gro
 * `['master_uri']` - SplunkAPI URI of the cluster master (Required for servers connecting to managed clusters)
 * `['settings']` -  Hash of Cluster settings (Required for servers connecting to managed clusters),
 * `['settings'][???]` - Valid values are those under the clustering stanza of [server.conf][]
+* `['settings'][???]['_cerner_splunk_indexer_count']` - The number of indexers to use for calculating maxTotalDataSizeMb for each index in combination with _maxDailyDataSizeMB in the index configuration.
```
Comment (nmattam) [inline docs/databags.md]: nit: _maxTotalDataSizeMB_

File: recipes/_configure_indexes.rb (hunk 1 above)
Comment (nmattam) [inline recipes/_configure_indexes.rb]: Padding added to ingestion per day to account for the metadata. If the ingestion per day is 50MB and if we decided to add 10% padding, _maxDailyDataSizeMB should be 55MB.

Comment (gravesb) [inline recipes/_configure_indexes.rb]: Fair point.  Added in a configurable padding here: https://github.com/cerner/cerner_splunk/pull/123/commits/654b2d5bbbcbad2e48d2e172637bf4c1d5cd17f0

File: recipes/_configure_indexes.rb (hunk 2 above)
Comment (gravesb) [inline recipes/_configure_indexes.rb]: Sure, changed here: https://github.com/cerner/cerner_splunk/pull/123/files#diff-79be619f4d42bccfc1205531edfcace2R49

File: docs/databags.md (hunk 3 above)
Comment (gravesb) [inline docs/databags.md]: Fixed in https://github.com/cerner/cerner_splunk/pull/123/files#diff-a8cd9604d38ad43b8b92ae521a075a6aR21

**Comment by jdmurphy** [comment]
+1

**Comment by nmattam** [comment]
+1

**Comment by acharlieh** [comment]
+1

**Pull request merged**